conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,431 @@
+    // @output error("Message",n=null)
+
+    io:println(() == let null n = null in <()>n); // @output true
+
+    io:println(let null n = null in typeof n); // @output typedesc ()
+
+    io:println(let null n = null in !(n == ())); // @output false
+
+    io:println(let null n = null in n is null); // @output true
+
+    io:println(let null n = null in n == null); // @output true
+    io:println(let null n = null in n != null); // @output false
+    io:println(let null n = null in n === null); // @output true
+    io:println(let null n = null in n !== null); // @output false
+
+    io:println(let null n = null in n == () ? true : false); // @output true
+
+    var a = let null n = null in from var i in [1, n, 2, n] select i;
+    io:println(a); // @output [1,null,2,null]
+    var b = let null n1 = null, null n2 = null in from var i in [1, n1, 2, n2] select i;
+    io:println(b); // @output [1,null,2,null]
+    var c = let null n = null in from var i in [1, n, 2, n]
+                                let int? x = i
+                                select x;
+    io:println(c); // @output [1,null,2,null]
+    var d = let null n = null in from var x in [
+                                                {id: 1, name: "X"},
+                                                {id: 2, name: n}
+                                            ]
+                                join var y in [
+                                                {id: 1, name: "Y"},
+                                                {id: 3, name: n}
+                                            ]
+                                on x.name equals y.name
+                                select x.id.toString() + " " + y.id.toString();
+    io:println(d); // @output ["2 3"]
+}
+
+function trueFunc() returns boolean {
+    return true;
+}
+
+function isNil(int? i) returns boolean {
+    return i == ();
+}
+
+Test-Case: output
+Description: Test let-expr, when type-descriptor is a boolean-type-descriptor which is a simple-type-descriptor.
+Labels: let-expr, module-class-defn, return-stmt, union-type, boolean, record-type, list-constructor-expr,
+        mapping-constructor-expr, table-constructor-expr, object-constructor-expr, new-expr, function-call-expr,
+        method-call-expr
+
+type Record record {|
+    int|boolean? i;
+|};
+
+class BoolClass {
+    boolean b1;
+    boolean b2;
+
+    function init(boolean b1, boolean b2) {
+        self.b1 = b1;
+        self.b2 = b2;
+    }
+
+    function foo() returns boolean {
+        return self.b1 && !self.b2;
+    }
+}
+
+function init() {
+    io:println(let boolean b = true in b); // @output true
+
+    io:println(let boolean b1 = true, boolean b2 = false in [b1, b2, b1]); // @output [true,false,true]
+    io:println(let boolean b1 = true, boolean b2 = false in [b2, [b1, b2]]); // @output [false,[true,false]]
+
+    io:println(let boolean b1 = true, boolean b2 = false in {"b": b1, b1: b1, b2, ["a"]: [b1, b2]});
```

Comment:
let's have the output in the same line as the print. If not, might impact future improvements in the test runner.

Reply:
I think that should not be affected. It should be able to add the output in the next line. Otherwise, it can exceed 120 characters per line.

Reply:
I agree with Lasini. The pattern we've followed so far is to have a line of code associated with the first output comment (or only comment if a single comment), I think we should continue doing that.

IMO, exceeding line length isn't much of an issue here, since once written we would expect the tests to change minimally or not at all. On the other hand, being able to match line of code vs output may add more value.